Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp

```diff
@@ -69,7 +69,7 @@ void DownloadProxyMap::platformDestroy()
 }
 #endif
 
-DownloadProxy& DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataStore, Ref<API::DownloadClient>&& client, const WebCore::ResourceRequest& resourceRequest, const FrameInfoData& frameInfo, WebPageProxy* originatingPage)
+Ref<DownloadProxy> DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataStore, Ref<API::DownloadClient>&& client, const WebCore::ResourceRequest& resourceRequest, const FrameInfoData& frameInfo, WebPageProxy* originatingPage)
```

Comment:
I thought the point here was that the download proxy was kept alive by the map and so the caller didn’t need a reference to it. But I suppose that’s too tricky to reason about, and this is better for security anyway.

Reply:
Yes, you are right about the ownership model. I still made the change since we prefer to use smart pointer more nowadays, even when not strictly needed.